src/Application/Application.php

```diff
@@ -117,14 +117,14 @@ public function createInitialRequest()
 			throw new BadRequestException('No route for HTTP request.');
 
 		} elseif (strcasecmp($request->getPresenterName(), $this->errorPresenter) === 0) {
-			throw new BadRequestException('Invalid request. Presenter is not achievable.');
+			throw (new BadRequestException('Invalid request. Presenter is not achievable.'))->setRequest($request);
 		}
 
 		try {
 			$name = $request->getPresenterName();
 			$this->presenterFactory->getPresenterClass($name);
 		} catch (InvalidPresenterException $e) {
-			throw new BadRequestException($e->getMessage(), 0, $e);
+			throw (new BadRequestException($e->getMessage(), 0, $e))->setRequest($request);
```

Comment:
:+1: for Doctrine style. Or we can at least remove `setRequest()` and pass it as 4th constructor's parameter. Exceptions should be immutable.